### PR TITLE
Simplify intrinsic dimension APIs

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/4178

File: packages/flutter/lib/src/rendering/flow.dart [hunk 1]
```diff
@@ -240,24 +240,40 @@ class RenderFlow extends RenderBox
...
   }
 
   @override
-  double getMinIntrinsicHeight(BoxConstraints constraints) {
-    return _getSize(constraints).height;
+  double getMinIntrinsicHeight(double width) {
+    final double height = _getSize(new BoxConstraints.tightForFinite(width: width)).height;
+    if (height.isFinite)
+      return height;
+    return 0.0;
```
Comment (abarth) [inline packages/flutter/lib/src/rendering/flow.dart]: Seems like we repeat this branch in a bunch of places.  We could pull it out into a function.

File: packages/flutter/test/rendering/aspect_ratio_test.dart [hunk 2]
```diff
@@ -4,21 +4,96 @@
 
 import 'package:flutter/rendering.dart';
 import 'package:test/test.dart';
+import 'rendering_tester.dart';
 
 void main() {
-  test('Intrinsic sizing', () {
+  test('Intrinsic sizing 2.0', () {
```
Comment (abarth) [inline packages/flutter/test/rendering/aspect_ratio_test.dart]: 2.0?

**Comment by abarth** [comment]
LGTM

**Pull request merged**